File: lint-arwaky/archive/contract/dispatch_commands_aggregate.py
"""dispatch_commands_aggregate — Command catalog moved from surfaces to contract aggregate.

This is the source of truth for available CLI/MCP commands. Both
infrastructure (for tool schema generation) and surfaces (for dispatch)
import from here.
"""

from ..taxonomy import ActionName, CommandMetadataVO

COMMAND_CATALOG: dict[ActionName, CommandMetadataVO] = {
    ActionName(value="check"): {
        "description": "Run full architecture compliance analysis",
        "example": "auto-lint check /path",
    },
    ActionName(value="scan"): {
        "description": "Deep directory scan (alias for check)",
        "example": "auto-lint scan ./src/",
    },
    ActionName(value="fix"): {"description": "Apply safe fixes", "example": "auto-lint fix file.py"},
    ActionName(value="report"): {
        "description": "Generate quality reports",
        "example": "auto-lint report ./src --format json",
    },
    ActionName(value="ci"): {
        "description": "CI-optimized with exit codes",
        "example": "auto-lint ci /path --exit-zero",
    },
    ActionName(value="batch"): {
        "description": "Check multiple paths",
        "example": "auto-lint batch path1.py path2.js",
    },
    ActionName(value="watch"): {
        "description": "Watch files for changes",
        "example": "auto-lint watch ./src/",
    },
    ActionName(value="security"): {
        "description": "Bandit vulnerability scanning",
        "example": "auto-lint security /path",
    },
    ActionName(value="complexity"): {
        "description": "Cyclomatic complexity",
        "example": "auto-lint complexity ./src/",
    },
    ActionName(value="duplicates"): {
        "description": "Code duplication detection",
        "example": "auto-lint duplicates /path",
    },
    ActionName(value="trends"): {
        "description": "Quality trend over time",
        "example": "auto-lint trends .",
    },
    ActionName(value="dependencies"): {
        "description": "Dependency vulnerability scan",
        "example": "auto-lint dependencies .",
    },
    ActionName(value="diff"): {
        "description": "Compare two versions",
        "example": "auto-lint diff v1.py v2.py",
    },
    ActionName(value="suggest"): {
        "description": "AI-powered suggestions",
        "example": "auto-lint suggest file.py",
    },
    ActionName(value="stats"): {
        "description": "Statistics dashboard",
        "example": "auto-lint stats ./src/",
    },
    ActionName(value="init"): {"description": "Initialize config", "example": "auto-lint init /path"},
    ActionName(value="config"): {
        "description": "Edit configuration",
        "example": "auto-lint config get thresholds",
    },
    ActionName(value="ignore"): {
        "description": "Manage ignore rules",
        "example": "auto-lint ignore add E501",
    },
    ActionName(value="import"): {
        "description": "Import configurations",
        "example": "auto-lint import config.json",
    },
    ActionName(value="export"): {
        "description": "Export reports",
        "example": "auto-lint export --format sarif",
    },
    ActionName(value="clean"): {"description": "Cleanup cache", "example": "auto-lint clean"},
    ActionName(value="update"): {"description": "Update adapters", "example": "auto-lint update"},
    ActionName(value="doctor"): {"description": "Diagnose issues", "example": "auto-lint doctor"},
    ActionName(value="adapters"): {
        "description": "List enabled adapters",
        "example": "auto-lint adapters",
    },
    ActionName(value="install-hook"): {
        "description": "Install git pre-commit hook",
        "example": "auto-lint install-hook",
    },
    ActionName(value="uninstall-hook"): {
        "description": "Remove git pre-commit hook",
        "example": "auto-lint uninstall-hook",
    },
    ActionName(value="cancel"): {
        "description": "Cancel a running lint job",
        "example": "auto-lint cancel <job_id>",
    },
    ActionName(value="plugins"): {
        "description": "List discovered and registered plugins",
        "example": "auto-lint plugins",
    },
    ActionName(value="multi-project"): {
        "description": "Run lint across multiple projects",
        "example": "auto-lint multi-project proj1/ proj2/",
    },
    ActionName(value="version"): {"description": "Show version", "example": "auto-lint version"},
}

File: lint-arwaky/src-rust/contract/dispatch_commands_aggregate.rs
use std::collections::HashMap;
use super::*;

pub fn command_catalog() -> HashMap<String, CommandMetadataVO> {
    let mut catalog = HashMap::new();
    catalog.insert("check".into(), CommandMetadataVO::new(DescriptionVO::new("Run full architecture compliance analysis"), Suggestion::new("auto-lint check /path")));
    catalog.insert("scan".into(), CommandMetadataVO::new(DescriptionVO::new("Deep directory scan"), Suggestion::new("auto-lint scan ./src/")));
    catalog.insert("fix".into(), CommandMetadataVO::new(DescriptionVO::new("Apply safe fixes"), Suggestion::new("auto-lint fix file.py")));
    catalog.insert("report".into(), CommandMetadataVO::new(DescriptionVO::new("Generate quality reports"), Suggestion::new("auto-lint report ./src --format json")));
    catalog.insert("ci".into(), CommandMetadataVO::new(DescriptionVO::new("CI-optimized with exit codes"), Suggestion::new("auto-lint ci /path --exit-zero")));
    catalog.insert("batch".into(), CommandMetadataVO::new(DescriptionVO::new("Check multiple paths"), Suggestion::new("auto-lint batch path1.py path2.js")));
    catalog.insert("watch".into(), CommandMetadataVO::new(DescriptionVO::new("Watch files for changes"), Suggestion::new("auto-lint watch ./src/")));
    catalog.insert("security".into(), CommandMetadataVO::new(DescriptionVO::new("Bandit vulnerability scanning"), Suggestion::new("auto-lint security /path")));
    catalog.insert("complexity".into(), CommandMetadataVO::new(DescriptionVO::new("Cyclomatic complexity"), Suggestion::new("auto-lint complexity ./src/")));
    catalog.insert("duplicates".into(), CommandMetadataVO::new(DescriptionVO::new("Code duplication detection"), Suggestion::new("auto-lint duplicates /path")));
    catalog.insert("trends".into(), CommandMetadataVO::new(DescriptionVO::new("Quality trend over time"), Suggestion::new("auto-lint trends .")));
    catalog.insert("dependencies".into(), CommandMetadataVO::new(DescriptionVO::new("Dependency vulnerability scan"), Suggestion::new("auto-lint dependencies .")));
    catalog.insert("diff".into(), CommandMetadataVO::new(DescriptionVO::new("Compare two versions"), Suggestion::new("auto-lint diff v1.py v2.py")));
    catalog.insert("suggest".into(), CommandMetadataVO::new(DescriptionVO::new("AI-powered suggestions"), Suggestion::new("auto-lint suggest file.py")));
    catalog.insert("stats".into(), CommandMetadataVO::new(DescriptionVO::new("Statistics dashboard"), Suggestion::new("auto-lint stats ./src/")));
    catalog.insert("init".into(), CommandMetadataVO::new(DescriptionVO::new("Initialize config"), Suggestion::new("auto-lint init /path")));
    catalog.insert("config".into(), CommandMetadataVO::new(DescriptionVO::new("Edit configuration"), Suggestion::new("auto-lint config get thresholds")));
    catalog.insert("ignore".into(), CommandMetadataVO::new(DescriptionVO::new("Manage ignore rules"), Suggestion::new("auto-lint ignore add E501")));
    catalog.insert("export".into(), CommandMetadataVO::new(DescriptionVO::new("Export reports"), Suggestion::new("auto-lint export --format sarif")));
    catalog.insert("clean".into(), CommandMetadataVO::new(DescriptionVO::new("Cleanup cache"), Suggestion::new("auto-lint clean")));
    catalog.insert("update".into(), CommandMetadataVO::new(DescriptionVO::new("Update adapters"), Suggestion::new("auto-lint update")));
    catalog.insert("doctor".into(), CommandMetadataVO::new(DescriptionVO::new("Diagnose issues"), Suggestion::new("auto-lint doctor")));
    catalog.insert("adapters".into(), CommandMetadataVO::new(DescriptionVO::new("List enabled adapters"), Suggestion::new("auto-lint adapters")));
    catalog.insert("install-hook".into(), CommandMetadataVO::new(DescriptionVO::new("Install git pre-commit hook"), Suggestion::new("auto-lint install-hook")));
    catalog.insert("uninstall-hook".into(), CommandMetadataVO::new(DescriptionVO::new("Remove git pre-commit hook"), Suggestion::new("auto-lint uninstall-hook")));
    catalog.insert("cancel".into(), CommandMetadataVO::new(DescriptionVO::new("Cancel a running lint job"), Suggestion::new("auto-lint cancel <job_id>")));
    catalog.insert("plugins".into(), CommandMetadataVO::new(DescriptionVO::new("List discovered plugins"), Suggestion::new("auto-lint plugins")));
    catalog.insert("multi-project".into(), CommandMetadataVO::new(DescriptionVO::new("Run lint across multiple projects"), Suggestion::new("auto-lint multi-project proj1/ proj2/")));
    catalog.insert("version".into(), CommandMetadataVO::new(DescriptionVO::new("Show version"), Suggestion::new("auto-lint version")));
    catalog
}
